faults/common.go

```diff
@@ -0,0 +1,124 @@
+// Package faults provides common headers and client-side fault injection functionality.
+package faults
+
+import (
+	"fmt"
+	"log/slog"
+	"math/rand/v2"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type GetHeaderFn func(key string) string
+type ResumeFn func() (interface{}, error)
+type ResponseFn func(code int, message string) (interface{}, error)
+type SleepFn func(d time.Duration)
```

Comment:
1. please write doccomments for exported types
2. for new code, please use `any` over `interface{}`.

Reply:
Done.